OpenFeature/src/main/java/dev/openfeature/sdk/events/EventPublisher.kt

```diff
@@ -0,0 +1,53 @@
+package dev.openfeature.sdk.events
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.filterIsInstance
+import kotlinx.coroutines.launch
+import kotlin.reflect.KClass
+
+sealed class OpenFeatureEvents {
+    object ProviderReady : OpenFeatureEvents()
+    object CacheUpdated : OpenFeatureEvents()
+}
```

Comment:
nit: do we want to separate these into a separate file later? I assume we will have more events to handle.